### PR TITLE
KAA-1415: [C SDK] KAA_RUNTIME_KEY_GENERATION=OFF issue on posix

### PR DESCRIPTION
**Pull request by mlysenko**

### CODE REVIEW

**Comment by rasendubi** [comment]
Please, refactor. Copy-pasting code is bad

**Comment by mlysenko** [comment]
It was made intentionally, because in original file, `KAA_RUNTIME_KEY_GENERATION` option was appliable only to posix target.

**Comment by mlysenko** [comment]
@rasendubi please review refactored code

File: client/client-multi/client-c/CMakeLists.txt [hunk 1]
```diff
@@ -277,24 +277,15 @@ add_subdirectory(Modules/cppcheck)
...
-                            INSTALL_COMMAND  "./generate_rsa_keys" COMMENT "Generating RSA headers"
-                            CMAKE_ARGS -DKAA_SDK_DIR=${KAA_SDK_DIR}
-                            BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/kaa_keys)
-
-        externalproject_add_step(kaa_rsa_keygen "Copying generated files"
-                                 COMMAND mv -f "${CMAKE_CURRENT_BINARY_DIR}/kaa_keys/kaa_keys_gen.h"
-                                 "${KAA_SRC_FOLDER}/gen" DEPENDEES install)
-    endif()
+
+option(KAA_RUNTIME_KEY_GENERATION "Enable RSA key generation at runtime" ON)
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt]: Should only be present if platform is posix.

File: client/client-multi/client-c/CMakeLists.txt [hunk 2]
```diff
@@ -277,24 +277,15 @@ add_subdirectory(Modules/cppcheck)
...
+
+if(${KAA_PLATFORM} STREQUAL "posix" AND ${KAA_RUNTIME_KEY_GENERATION})
+    add_definitions(-DKAA_RUNTIME_KEY_GENERATION)
+    add_subdirectory(tools/kaa_encryption/rsa_key_gen)
+    target_link_libraries(kaac PUBLIC rsa_keygen)
 else()
+    set(KAA_RUNTIME_KEY_GENERATION OFF)
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt]: This has no effect

**Comment by rasendubi** [comment]
Please, don't merge without @Slash32's feedback

**Comment by mlysenko** [comment]
@rasendubi please review fixed version

**Comment by rasendubi** [comment]
https://github.com/mlysenko/kaa/pull/1

**Comment by mlysenko** [comment]
@rasendubi, accepted your PR, please merge.

**Comment by rasendubi** [comment]
Reverted the merge. Please, check Travis failure

**Comment by vadimol** [comment]
Could you please point me to the revert commit? I can't find it.

**Comment by rasendubi** [comment]
@vadimol there is no revert commit. I've force-push removed the commit in 5 seconds after merging.

**Pull request merged**